ACPs/77-reinventing-subnets/README.md

```diff
@@ -177,41 +272,60 @@ type ExitValidatorSetTx struct {
 }
 ```
 
-#### SetSubnetValidatorManagerTx
+### Proof of Subnet Validator Set Change
 
-To participate in this new registration flow, Subnets must set the `(blockchainID, address)` pair by using `SetSubnetValidatorManagerTx`. This pair will be used to validate the `AddressedCall` in `SetSubnetValidatorWeightTx` and `RegisterSubnetValidatorTx`.
+To track whether a Subnet Validator addition/modification/removal occured on the P-Chain, Subnet Validators must be willing to sign an `AddressedCall` attesting to the validator set change. Since all Subnet Validators sync the P-Chain, only the Subnet Validators need to sign the `AddressedCall`, not all Primary Network Validators.
 
-When a Subnet is created with `CreateSubnetTx`, a validator manager is not specified. To solve this problem, the first `SetSubnetValidatorManagerTx` will expect the Warp `AddressedCall` to have the `sourceChainID` be set to the `SubnetID` and the `sourceAddress` to be zero'd out. Once this transaction is accepted, `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` are enabled for use on the Subnet.
+Two `AddressedCall`s are defined with the below payloads. For each of them, the `sourceChainID` must be set to the P-Chain ID and the `sourceAddress` must be set to an empty byte array.
 
-```golang
-type SetSubnetValidatorManagerTx struct {
-    // Metadata, inputs and outputs
-    BaseTx
-    // Warp message should include:
-    //   - SubnetID
-    //   - ChainID (where the validator manager lives)
-    //   - Address (address of the validator manager)
-    //   - BLS multisig over the above payload
-    Message warp.Message `json:"message"`
-}
+The method of requesting is left unspecified as it can be more generic. A viable option for supporting this functionality is laid out in [ACP-118](../118-warp-signature-request/README.md) with the `SignatureRequest` message.
+
+#### SubnetValidatorRegistrationMessage
```

Comment:
I am wondering if having 2 payloads is really needed.  
Can't we always use the weight update payload below? Even to check for successful registration with `nonce` = `0`.

Reply:
I don't think it's strictly needed. However, I think it's cleaner to have two separate ones to correspond to the two different warp messages used in `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx`.